python/pyarrow/tests/test_table.py

```diff
@@ -2302,6 +2302,9 @@ def test_table_filter_expression():
         "colVals": ["a", "b", "f", "B", "A"]
     })
 
+    with pytest.raises(NotImplementedError):
+        result = t1.filter(t2)
```

Comment:
The tests need not depend on Acero, perhaps it should be added to `test_compute.py` instead?